qcodes/instrument/meta/meta_instrument.py

```diff
@@ -0,0 +1,185 @@
+from typing import List, Dict, Union, Any, Optional
+
+from functools import partial
+
+from qcodes.instrument.meta.meta_parameter import MetaGroup, MetaParameter
+from qcodes.instrument.parameter import Parameter
+from qcodes.instrument.base import InstrumentBase
+from qcodes.station import Station
+
+import logging
+
+_log = logging.getLogger(__name__)
+
+
+class MetaInstrument(InstrumentBase):
+    """MetaInstrument class for creating a meta instrument that aliases one or more 
+    parameter endpoints from real instruments.
+
+    Example usage in instrument YAML:
+
+    field:
+        type: qcodes.instrument.meta.MetaInstrument
+        init:
+        aliases:
+            X:
+            - field_X.field
+            ramp_rate:
+            - field_X.ramp_rate
+        set_initial_values_on_load: true
+        initial_values:
+            ramp_rate: 0.02
+        setters:
+            X:
+            method: field_X.set_field
+            block: false
+        units:
+            X: T
+            ramp_rate: T/min
+
+    Args:
+        name: Instrument name
+        station: Real instrument station to connect the endpoints to.
+        aliases: Aliases and the endpoints they connect to.
+        initial_values: Default values to set on meta instrument.
+            Defaults to None.
+        set_initial_values_on_load: Flag to set defaults on load or not. Defaults to False.
+        setters: Optional setter methods to use instead
+            of calling the .set() method on the endpoint parameters. Defaults to None.
+        metadata: Optional metadata to pass to instrument. Defaults to None.
+    """
+    param_cls = MetaParameter
+
+    @staticmethod
+    def parse_instrument_path(station: Station, path: Union[str, List[str]]):
+        """Parse a string path and return the object relative to the station,
+        e.g. "my_instrument.my_param" returns station.my_instrument.my_param
+
+        Args:
+            station: Measurement station
+            path: Relative path to parse
+        """
+        def _parse_path(parent, elem):
+            child = getattr(parent, elem[0])
+            if len(elem) == 1:
+                return child
+            return _parse_path(child, elem[1:])
+
+        return _parse_path(station, path.split("."))
```

Comment:
I find that putting the static method between the init method and the docstring significantly reduces readability since one has to scroll to compare the docstring with the signature